core/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1083,53 +1084,40 @@ public void onMessage(ITransport transport, ProtocolMessage message) throws Ably
     }
 
     private void onChannelMessage(ProtocolMessage message) {
-        if(message.connectionSerial != null) {
-            connection.serial = message.connectionSerial.longValue();
-            if (connection.key != null)
-                connection.recoveryKey = connection.key + ":" + message.connectionSerial;
-        }
         channels.onMessage(message);
     }
 
     private synchronized void onConnected(ProtocolMessage message) {
-        /* if the returned connection id differs from
-         * the existing connection id, then this means
-         * we need to suspend all existing attachments to
-         * the old connection.
-         * If realtime did not reply with an error, it
-         * signifies that this was a result of an earlier
-         * connection being invalidated due to being stale.
-         *
-         * Suspend all channels attached to the previous id;
-         * this will be reattached in setConnection() */
         ErrorInfo error = message.error;
-        if(connection.id != null && !message.connectionId.equals(connection.id)) {
-            /* we need to suspend the original connection */
-            if(error == null) {
-                error = REASON_SUSPENDED;
-            }
-            channels.suspendAll(error, false);
-        }
 
-        /* set the new connection id */
-        ConnectionDetails connectionDetails = message.connectionDetails;
-        connection.key = connectionDetails.connectionKey;
-        if (!message.connectionId.equals(connection.id)) {
-            /* The connection id has changed. Reset the message serial and the
-             * pending message queue (which fails the messages currently in
-             * there). */
-            pendingMessages.reset(msgSerial,
-                    new ErrorInfo("Connection resume failed", 500, 50000));
+        if (message.action == ProtocolMessage.Action.connected && message.connectionId.equals(connection.id) && error == null) {
+            //RTN15c6
+            Log.d(TAG, "connection has reconnected and resumed successfully");
+            connection.reason = null;
+            channels.reAttach();
+
+            //RTN19a
+            if (pendingMessages.queue != null && !pendingMessages.queue.isEmpty()) {
```

Comment:
Should the messages be processed in both branches given in both cases we have a connection? `RTN15c7` also says:
> The connection should also process any messages queued per @RTL6c2@.

In which case should this block be moved outside the `if/else` too?

Reply:
So we are good here?

Reply:
If `State.enact` already implements this then shouldn't this block (sending `pendingMessages`) be removed?

Though I'm not really familiar with the sdk so may be misunderstanding - if its not already implemented this should be sending `pendingMessages` in both branches

Reply:
I moved this spec to the outside of if/else but I am not sure if I should remove a message from the pending queue once it is sent as this implementation for sending is not clear.